src/t_stream.c

```diff
@@ -3373,16 +3443,32 @@ void xautoclaimCommand(client *c) {
             return; /* Type error. */
         group = streamLookupCG(o->ptr,c->argv[2]->ptr);
     }
-
-    /* No key or group? Send an error given that the group creation
-     * is mandatory. */
-    if (o == NULL || group == NULL) {
-        addReplyErrorFormat(c,"-NOGROUP No such key '%s' or consumer group '%s'",
-                            (char*)c->argv[1]->ptr,
-                            (char*)c->argv[2]->ptr);
+    if(o == NULL) {
+        addReplyErrorFormat(c, "No such key '%s'",
+                                     (char*)c->argv[1]->ptr);
         return;
     }
 
+    if (group == NULL) {
+        if (!mkgroup) {
+           addReplyErrorFormat(c, "-NOGROUP No consumer "
+                                  "group '%s'", (char*)c->argv[2]->ptr);
+           return;
+        }
+        streamID id;
+        id.ms = 0;
+        id.seq = 0;
+        group = streamCreateCG(o->ptr,c->argv[2]->ptr,strlen(c->argv[2]->ptr),&id,SCG_INVALID_ENTRIES_READ);
+        if(group) {
+            server.dirty++;
+            notifyKeyspaceEvent(NOTIFY_STREAM,"xgroup-create",c->argv[1],c->db->id);
+        } else {
+            addReplyErrorFormat(c, "Create consumer group '%s' in CAUTOLAIM failed" , (char*)c->argv[2]->ptr);
+            return;
```

Comment:
drop impossible error

Reply:
I will update the codes following your comments, and rebase it as well.  Thanks Oran